src/DebuggerForm.cpp

```diff
@@ -475,6 +487,25 @@ void DebuggerForm::createToolbars()
 	executeToolbar->addAction(executeStepOutAction);
 	executeToolbar->addAction(executeStepBackAction);
 	executeToolbar->addAction(executeRunToAction);
+
+	// create customised toolbar
+	userToolbar = addToolBar(tr("User defined"));
+	userToolbar->setEnabled(false);
+	connect(&comm, &CommClient::connectionReady, userToolbar, [this]{ userToolbar->setEnabled(true); });
+	connect(&comm, &CommClient::connectionTerminated, userToolbar, [this]{ userToolbar->setEnabled(false); });
+	updateCustomActions();
+}
+
+void DebuggerForm::updateCustomActions()
+{
+	userToolbar->clear();
+	for (const auto& command: commands) {
```

Comment:
Detail (style): in most of the code we put a space both before and after the `:`.
This comment applies to several places in this commit.